### DISCUSSION TITLE
请问new_kcp版本和development版本有什么区别？What is the difference between the two versions?

### DISCUSSION THREAD

**Discussion by daxiwow** [Melledy/Grasscutter#1253] [Q&A] [answered]
<img width="306" alt="捕获" src="https://user-images.githubusercontent.com/54258434/173344644-c45ceea1-ebd6-49ce-91c1-8697035d716f.PNG">

**Comment by daxiwow** [comment]
What is the difference between the two versions?

**Comment by zhaodice** [comment, marked as answer]
The new KCP version is that I replaced the old KCP Library (no one maintained) at the bottom of the mower with a new KCP library. It uses epoll engine, which is used for millions of concurrency. Moreover, the new KCP library has been adopted by Tencent, Kwai and other projects. I think the performance is guaranteed.
detail: https://github.com/Grasscutters/Grasscutter/pull/1237

new-kcp版本是我把割草机的底层的旧kcp库（没人维护了）换成了新的kcp库，它采用了epoll引擎，该引擎库用于百万并发，而且该新kcp库被腾讯、快手等多个项目所采用，我觉得性能是有保障的。
详情：https://github.com/Grasscutters/Grasscutter/pull/1237
